src/friendly-errors-plugin.js

```diff
@@ -110,6 +111,27 @@ class FriendlyErrorsWebpackPlugin {
     formatErrors(topErrors, this.formatters, severity)
       .forEach(chunk => output.log(chunk));
   }
+
+  getCompileTime(stats, statsIndex) {
+    if (isMultiStats(stats)) {
+      // Webpack multi compilations run in parallel so using the longest duration.
+      // https://webpack.github.io/docs/configuration.html#multiple-configurations
+      return stats.stats
+        .reduce((time, stats, index) => Math.max(time, this.getCompileTime(stats, index)), 0);
+    }
+
+    // When we have multi compilations but only one of them is rebuilt, we need to skip the
+    // unchanged compilers to report the true rebuild time.
+    if (statsIndex !== undefined) {
```

Comment:
Everything below the multiStat check can probably be moved into another function, so that no recursion happens when the stat is multiStat. That's both less error-prone and easier to follow for people reading the code.